lib/rouge/lexers/sas.rb

```diff
@@ -0,0 +1,568 @@
+          HPFMM HPGENSELECT HPIMPUTE HPLMIXED HPLOGISTIC HPMIXED HPNLMOD
+          HPPANEL HPPLS HPPRINCOMP HPQUANTSELECT HPQLIM HPREG HPSAMPLE
+          HPSEVERITY HPSPLIT HPSUMMARY HTSNP HTTP ICLIFETEST ICPHREG IML IMPORT
+          IMSTAT IMXFER INBREED INFOMAPS INTPOINT IOMOPERATE IRT ISHIKAWA ITEMS
+          JAVAINFO JSON KDE KRIGE2D LASR LATTICE LIFEREG LIFETEST LOAN
+          LOCALEDATA LOESS LOGISTIC LP LUA MACONTROL MAPIMPORT MCMC MDC MDDB
+          MDS MEANS METADATA METALIB METAOPERATE MI MIANALYZE MIGRATE MIXED
+          MODECLUS MODEL MSCHART MULTTEST MVPDIAGNOSE MVPMODEL MVPMONITOR
+          NESTED NETDRAW NETFLOW NLIN NLMIXED NLP NPAR1WAY ODSLIST ODSTABLE
+          ODSTEXT OLAP OLAPCONTENTS OLAPOPERATE OPERATE OPTEX OPTGRAPH OPTIONS
+          OPTLOAD OPTLP OPTLSO OPTMILP OPTMODEL OPTNET OPTQP OPTSAVE ORTHOREG
+          PANEL PARETO PDLREG PDS PDSCOPY PHREG PLAN PLM PLOT PLS PM PMENU
+          POWER PRESENV PRINCOMP PRINQUAL PRINT PRINTTO PROBIT PROTO PRTDEF
+          PRTEXP PSMOOTH PWENCODE QDEVICE QLIM QUANTLIFE QUANTREG QUANTSELECT
+          QUEST RANK RAREEVENTS RDC RDPOOL RDSEC RECOMMEND REG REGISTRY RELEASE
+          RELIABILITY REPORT RISK ROBUSTREG RSREG SCAPROC SCORE SEQDESIGN
+          SEQTEST SERVER SEVERITY SGDESIGN SGPANEL SGPLOT SGRENDER SGSCATTER
+          SHEWHART SIM2D SIMILARITY SIMLIN SIMNORMAL SOAP SORT SOURCE SPECTRA
+          SPP SQL SQOOP SSM STANDARD STATESPACE STDIZE STDRATE STEPDISC STP
+          STREAM SUMMARY SURVEYFREQ SURVEYIMPUTE SURVEYLOGISTIC SURVEYMEANS
+          SURVEYPHREG SURVEYREG SURVEYSELECT SYSLIN TABULATE TAPECOPY TAPELABEL
+          TEMPLATE TIMEDATA TIMEID TIMEPLOT TIMESERIES TPSPLINE TRANSPOSE
+          TRANSREG TRANTAB TREE TSCSREG TTEST UCM UNIVARIATE UPLOAD VARCLUS
+          VARCOMP VARIOGRAM VARMAX VASMP X11 X12 X13 XSL
+        )
+      end
+
+      state :basics do
+        # Rules to be parsed before the keywords (which are different depending
+        # on the context)
+
+        rule /\s+/m, Text
+
+        # Single-line comments (between * and ;) - these can actually go onto multiple lines
+        # case 1 - where it starts a line
+        rule /^\s*%?\*[^;]*;/m, Comment::Single
+        # case 2 - where it follows the previous statement on the line (after a semicolon)
+        rule /(;)(\s*)(%?\*[^;]*;)/m do
+          groups Punctuation, Text, Comment::Single
+        end
+
+        # True multiline comments!
+        rule %r(/[*].*?[*]/)m, Comment::Multiline
+
+        # date/time constants (Language Reference pp91-2)
+        rule /'[0-9a-z]+?'d/i, Literal::Date
+        rule /'.+?'dt/i, Literal::Date
+        rule /'[0-9:]+?([a|p]m)?'t/i, Literal::Date
+
+        rule /'/, Str::Single, :single_string
+        rule /"/, Str::Double, :double_string
+        rule /&[a-z0-9_&.]+/i, Name::Variable
+
+        # numeric constants (Language Reference p91)
+        rule /\d[0-9a-f]*x/i, Num::Hex
```

Comment:
Hexadecimal numbers couldn't start with a letter? And can you add some examples of these to your visual sample?

Reply:
Yes, SAS requires a digit at the beginning, to distinguish the number from a variable name - so to write the decimal number 10 in hex you have to say `0ax`. I've added an example in https://github.com/rouge-ruby/rouge/pull/1107/commits/3191ee40b262959258188a1c4661249f0daf3520